contrib/raftexample/raftexample_test.go

```diff
@@ -205,3 +205,41 @@ func TestPutAndGetKeyValue(t *testing.T) {
 		t.Fatalf("expect %s, got %s", wantValue, gotValue)
 	}
 }
+
+// TestAddNewNode tests adding new node to the existing cluster.
+func TestAddNewNode(t *testing.T) {
+	clus := newCluster(3)
+	defer clus.closeNoErrors(t)
+
+	os.RemoveAll("raftexample-4")
+	os.RemoveAll("raftexample-4-snap")
+	defer func() {
+		os.RemoveAll("raftexample-4")
+		os.RemoveAll("raftexample-4-snap")
+	}()
+
+	newNodeURL := "http://127.0.0.1:10004"
+	clus.confChangeC[0] <- raftpb.ConfChange{
+		Type:    raftpb.ConfChangeAddNode,
+		NodeID:  4,
+		Context: []byte(newNodeURL),
+	}
+
+	proposeC := make(chan string)
+	defer close(proposeC)
+
+	confChangeC := make(chan raftpb.ConfChange)
+	defer close(confChangeC)
+
+	commitC, _, _ := newRaftNode(4, append(clus.peers, newNodeURL), true, nil, proposeC, confChangeC)
+
+	<-time.After(time.Second * 3)
```

Comment:
Waiting on real time in tests is a bad idea.
 
Could you please wait till whatever you need happens (in worse case with 10ms retry loop), 
such that test-cases can finish as fast as possible. Otherwise we have unit tests that needs minutes to finish whole suite.

Reply:
Changed to propose to new node and check commitC in the existing cluster. This pattern doesn't require wait.